### PR TITLE
feat: add counts while getting doc gen

### PR DESCRIPTION
**Pull request by svipulrao98**
### Problem

Send the column Counts while getting the Doc Gen.

### Solution

Describe the implemented solution. Add external references if needed.

### Screenshot/Demo

A picture is worth a thousand words. Please highlight the changes if applicable.

### How to test

- Steps to be followed to verify the solution or code changes
- Mention if there is any settings configuration added/changed/deleted

## Checklist

- [ ] I have run this code and it appears to resolve the stated issue
- [ ] `README.md` updated and added information about my change

### CODE REVIEW

File: src/services/docGenService.ts [hunk 1]
```diff
@@ -107,6 +110,7 @@ export class DocGenService {
                 adapter,
                 message,
                 columns,
+                1,
```
Comment (saravmajestic) [inline src/services/docGenService.ts:113]: this should be columnCount?

File: src/altimate.ts [hunk 2]
```diff
@@ -170,6 +170,7 @@ interface DocsGenerateModelRequestV2 {
   };
   prompt_hint?: string;
   gen_model_description: boolean;
+  columns_count: number;
```
Comment (saravmajestic) [inline src/altimate.ts:173]: variable naming could be improved. `current_column_index` or `completed_columns_count`?

Comment (suryaiyer95) [reply src/altimate.ts:173]: Agree

File: src/services/docGenService.ts (hunk 1 above)
Comment (suryaiyer95) [reply src/services/docGenService.ts:113]: ColumnIndex right?

Comment (suryaiyer95) [reply src/services/docGenService.ts:113]: actually no. It should be completedColumns count and is applicable only for doc gen

File: src/altimate.ts (hunk 2 above)
Comment (suryaiyer95) [reply src/altimate.ts:173]: I think along with this, we send an unique id. session_id.table_id.bulk_gen or somthing like that. 

https://github.com/AltimateAI/vscode-dbt-power-user/pull/1151/files#diff-f5d2312039a56f9d5d5c2eff4188b2ab44446ae157202ee46a51da2a5ef242c7R16

You can find it here

Comment (suryaiyer95) [reply src/altimate.ts:173]: You can store this in the backend and say (2000 executions) charged for (300) or smthg like that

Comment (suryaiyer95) [reply src/altimate.ts:173]: vscode_session_id-{model-unique-id}-{bulk.gen}

File: src/services/docGenService.ts [hunk 3]
```diff
@@ -402,6 +416,8 @@ export class DocGenService {
                   message.user_instructions.prompt_hint || "generate",
               },
               follow_up_instructions: message.follow_up_instructions,
+              column_index_count: columnIndexCount,
```
Comment (suryaiyer95) [inline src/services/docGenService.ts:419]: @saravmajestic When retry happens due to rate limiting, the columnIndexCount does not change right?

Comment (saravmajestic) [reply src/services/docGenService.ts:429]: it wont change. it will pickup same index

**Review by saravmajestic — APPROVED** [review]

**Review by AdiGajbhiye — APPROVED** [review]
lgtm

**Pull request merged**